File: leetcode/705/705.py
class MyHashSet:

    def __init__(self):
        self.size = 10**6+1
        self.bucket = [[] for _ in range(self.size)]

    def _hash(self, n):
        return n % self.size

    def add(self, key: int) -> None:
        idx = self._hash(key)
        if not self.contains(key):
            self.bucket[idx].append(key)

    def remove(self, key: int) -> None:
        idx = self._hash(key)
        if self.contains(key):
            self.bucket[idx].remove(key)

    def contains(self, key: int) -> bool:
        idx = self._hash(key)
        return True if key in self.bucket[idx] else False


# Your MyHashSet object will be instantiated and called as such:
# obj = MyHashSet()
# obj.add(key)
# obj.remove(key)
# param_3 = obj.contains(key)

File: leetcode/705/705.go
package main

import "slices"

type MyHashSet struct {
	size   int
	bucket [][]int
}

func Constructor() MyHashSet {
	size := 100001
	bucket := make([][]int, size)
	return MyHashSet{
		size:   size,
		bucket: bucket,
	}
}

func (this *MyHashSet) hash(key int) int {
	return key % this.size
}

func (this *MyHashSet) Add(key int) {
	idx := this.hash(key)
	if !this.Contains(key) {
		this.bucket[idx] = append(this.bucket[idx], key)
	}
}

func (this *MyHashSet) Remove(key int) {
	idx := this.hash(key)
	i := slices.Index(this.bucket[idx], key)
	if i != -1 {
		this.bucket[idx] = slices.Delete(this.bucket[idx], i, i+1)
	}
}

func (this *MyHashSet) Contains(key int) bool {
	idx := this.hash(key)
	return slices.Contains(this.bucket[idx], key)
}

/**
 * Your MyHashSet object will be instantiated and called as such:
 * obj := Constructor();
 * obj.Add(key);
 * obj.Remove(key);
 * param_3 := obj.Contains(key);
 */
